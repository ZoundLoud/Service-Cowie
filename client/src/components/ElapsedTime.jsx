import React from 'react';

const ElapsedTime = props => (
  <div className="Time-Container">
    <div className="Elapsed-time">
        2 Years Ago
    </div>
    <button className="hashtag" type="submit"># COWIE</button>
  </div>
);

export default ElapsedTime;
