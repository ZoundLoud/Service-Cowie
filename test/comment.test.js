import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount, shallow } from 'enzyme';
import ElapsedTime from '../client/src/components/ElapsedTime';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('Comment Component', () => {
  it('should render Comment without throwing an error', () => {
    const props = {
      imgurl: 'https://s3-us-west-1.amazonaws.com/songimages/photos/simg100.jpg',
      username: 'Annabelle Parker',
      comment: 'Nihil rerum possimus suscipit architecto non consequuntur provident adipisci.',
    };
    const songcomment = shallow(<new Comment random={props.imgurl} info={props.comment} name={props.username} />);
    expect(toJson(songcomment)).toMatchSnapshot();
  });
  it('should render correctly in "debug" mode', () => {
    const props = {
      imgurl: 'https://s3-us-west-1.amazonaws.com/songimages/photos/simg100.jpg',
      username: 'Annabelle Parker',
      comment: 'Nihil rerum possimus suscipit architecto non consequuntur provident adipisci.',
    };
    const debug = shallow(<new Comment random={props.imgurl} info={props.comment} name={props.username} debug />);
  
    expect(toJson(debug)).toMatchSnapshot();
  });
});
