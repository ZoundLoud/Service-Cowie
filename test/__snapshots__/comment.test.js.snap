// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Component should render Comment without throwing an error 1`] = `
<new
  Comment={true}
  info="Nihil rerum possimus suscipit architecto non consequuntur provident adipisci."
  name="Annabelle Parker"
  random="https://s3-us-west-1.amazonaws.com/songimages/photos/simg100.jpg"
/>
`;

exports[`Comment Component should render correctly in "debug" mode 1`] = `
<new
  Comment={true}
  debug={true}
  info="Nihil rerum possimus suscipit architecto non consequuntur provident adipisci."
  name="Annabelle Parker"
  random="https://s3-us-west-1.amazonaws.com/songimages/photos/simg100.jpg"
/>
`;
